lib/board.ex

```diff
@@ -21,14 +32,34 @@ defmodule Board do
   def draw?(board), do: board_full?(board) && !winner?(board)
 
   def board_full?(board) do
-    [] == Enum.filter(board, fn(cell) -> cell != "X" && cell != "O" end)
+    board
+    |> Enum.reject(&mark?/1)
+    |> Enum.empty?
   end
 
+  def rows(board) do
+    Enum.chunk(board, 3)
+  end
+
+  defp mark?(mark), do: mark in ["X", "O"]
+
   def winning_mark(board), do: winning_mark(board, @winning_combinations)
 
-  def position_available?(position, board) do
+  def position_available?(board, position) do
     cell = Enum.at(board, position - 1)
-    !(cell == "X" || cell == "O")
+    if cell in ["X", "O"] do
```

Comment:
This could be `if mark?(cell) do` to reduce the number of places that directly call "X" or "O"